app/models/journeyDomain/transport/TransportMeansActiveDomain.scala

```diff
@@ -18,27 +18,47 @@ package models.journeyDomain.transport
 
 import cats.implicits._
 import models.Index
+import models.SecurityDetailsType.NoSecurityDetails
 import models.domain.{GettableAsFilterForNextReaderOps, GettableAsReaderOps, UserAnswersReader}
-import models.journeyDomain.JourneyDomainModel
 import models.reference.{CustomsOffice, Nationality}
 import models.transport.transportMeans.active.Identification
+import models.transport.transportMeans.departure.InlandMode
+import models.transport.transportMeans.departure.InlandMode.Air
+import pages.preTaskList.SecurityDetailsTypePage
 import pages.transport.transportMeans.active._
+import pages.transport.transportMeans.departure.InlandModePage
 
 case class TransportMeansActiveDomain(
   identification: Identification,
   identificationNumber: String,
   nationality: Option[Nationality],
-  customsOffice: CustomsOffice
-) extends JourneyDomainModel
+  customsOffice: CustomsOffice,
+  conveyanceReferenceNumber: Option[String]
+)
 
 object TransportMeansActiveDomain {
 
+  def conveyanceReads(index: Index): UserAnswersReader[Option[String]] = {
+
+    val details = for {
+      securityDetails <- SecurityDetailsTypePage.reader
+      inlandMode      <- InlandModePage.reader
+    } yield (securityDetails, inlandMode)
+
+    details.flatMap {
+      case (NoSecurityDetails, x: InlandMode) if x != Air =>
```

Comment:
Only comment I would have would be to name this something more meaningful than 'x' however, thats more of a preference thing.